src/server/modules/qr/QrCodeController.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from 'express'
+import { inject, injectable } from 'inversify'
+import { UrlRepositoryInterface } from '../../repositories/interfaces/UrlRepositoryInterface'
+
+import ImageFormat from '../../../shared/util/image-format'
+import { DependencyIds } from '../../constants'
+
+import { QrCodeService } from './interfaces'
+
+@injectable()
+export class QrCodeController {
+  private ogUrl: string
+
+  private qrCodeService: QrCodeService
+
+  private urlRepository: UrlRepositoryInterface
+
+  constructor(
+    @inject(DependencyIds.ogUrl) ogUrl: string,
+    @inject(DependencyIds.qrCodeService) qrCodeService: QrCodeService,
+    @inject(DependencyIds.urlRepository) urlRepository: UrlRepositoryInterface,
+  ) {
+    this.ogUrl = ogUrl
+    this.qrCodeService = qrCodeService
+    this.urlRepository = urlRepository
+  }
+
+  private shortUrlExists: (shortUrl: string) => Promise<boolean> = async (
+    shortUrl,
+  ) => {
+    return !!(await this.urlRepository.findByShortUrl(shortUrl))
```

Comment:
This will be the first time a controller calls a repository directly in our codebase. I'm getting [mixed responses](https://softwareengineering.stackexchange.com/questions/306890/is-it-bad-practice-that-controller-call-repository-instead-of-service) online whether this is a good practice.

Some factors complicate this on top of what was discussed on stackoverflow (will edit if i think of more):

1) This is really just an existence predicate. does it really need to be in a service?

Open to discussion on this matter, we can also talk about this in syncup

Reply:
As noted, there really isn't much biz logic to isolate from the controller. While it is true that this is slightly inconsistent from the controller -> service -> repository pattern that is common in this codebase, I don't think it sets a precedent for going directly to repositories.

Given also that this is existing code lifted from the route into its own module, I would lean towards keeping it for now, and refactoring it later as the need arises.